### ISSUE TITLE
QB Session

### ISSUE THREAD

**Issue by coreman000** [QuickBlox/quickblox-ios-sdk#311]
Hi.
I can't find friendly document which will explain me how does QBSession work. 
for example: user logs in with Facebook, and after this signs in to Quickblox using FBToken.
As i understand, at this moment session starts and User can make requests. 

When and under what circumstances session ends?
When the session ends, does current_user become equal nil? - and is this  the only clue to see if session ended?

Are there any possibilities to stop session programmatically, just to test - what if app awakes and the session is over?

**Comment by Raikerian** [comment]
Hi,

1. In a latest versions of our SDK session is being managed automatically.
2. currentUser will not become nil after session expiration.
3. If session has been expired, any QBRequest method will first renew it and then execute itself.
4. You can destroy session using 'destroySessionWithCompletionBlock' method of QBRequest.
5. In order to connect in chat when you have logged in using any social provider, user QBSession token as password ([QBSession currentSession].sessionDetails.token).

**Comment by coreman000** [comment]
so if, for example, user tries a [request] with no internet connection, he or she will receive an error response from [request] ?

**Comment by Raikerian** [comment]
Yes, you are right.

**Comment by coreman000** [comment]
so, once logged in, i don't need to think about session status. great!

**Comment by coreman000** [comment]
Is it possible that i find nil in "QBSession.currentSession().currentUser! " ? I'm just wondering, should i make all this Optional checking...
If currentUser is never nil after it is defined once, may be you could make it not Optional, but just Lazy...

**Comment by Raikerian** [comment]
It is possible when session was created but user is not defined yet (not logged in) first time app opened on a device.

**Comment by coreman000** [comment]
If I understand correctly, "Lazy" is invented exactly for this situation.

**Comment by iSevenDays** [comment]
Hi coreman000,

Session can be created without user, then currentSession().currentUser will be nil.

SDK sets currentUser to be QBUUser instance only after login with user.

If you always use session with user, you can use QBSession.currentSession().currentUser!

Thanks!

**Comment by Raikerian** [comment]
Thanks for you feedback, we will update our session documentation in a near future.

Do you have any other questions regarding session?

**Comment by Raikerian** [comment]
If you are still having any troubles please let us know by creating a separate issue :)

**Comment by YogendraPatel** [comment]
@Raikerian 

I have face a issue in login and signup in quickblox.
Got every time error

error: Request failed: unauthorized (401)
reasons: {
    errors =     {
        base =         (
            "Required session does not exist"
        );
    };
}

That means QBRequest not manage session automatically

Please help me as soon as possible

Thanks in advance
